src/rgw/rgw_admin.cc

```diff
@@ -6140,6 +6143,25 @@ int main(int argc, const char **argv)
   }
 
   if (opt_cmd == OPT_USER_STATS) {
+    if (user_id.empty()) {
+      cerr << "ERROR: uid not specified" << std::endl;
+      return EINVAL;
+    }
+
+    string user_str = user_id.to_str();
+    if (recalculate_stats) {
+      if (!bucket_name.empty()){
+	cerr << "ERROR: recalculate doesn't work on buckets" << std::endl;
+	return EINVAL;
+      }
+      ret = store->cls_user_clear_header(user_str);
+      if (ret < 0) {
+	cerr << "ERROR: could not clear user stats: " << cpp_strerror(-ret) << std::endl;
+	return -ret;
+      }
+      sync_stats = true;
```

Comment:
@oritwas I can drop the sync_stats here, so the user can call it with --reset-stats --sync-stats if they need to explicitly sync stats or should we make sure that reset stats are also in sync with what cluster sees

Reply:
you can drop it as you don't need it because you write the heade with the recalculated stats.